File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let count = 0;

    function dfs(depth, total) {
        if (depth === numbers.length) {
            if (total === target) count += 1;
            return;
        }
        dfs(depth + 1, total + numbers[depth]);
        dfs(depth + 1, total - numbers[depth]);
    }

    dfs(0, 0);
    return count;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py


def solution(numbers, target):
    count = 0
    
    def dfs(depth, total):
        nonlocal count
        if depth == len(numbers):
            if total == target:
                count += 1
            return 
    
        dfs(depth + 1, total + numbers[depth])
        dfs(depth + 1, total - numbers[depth])
        
    dfs(0,0)
    return count
    